commons/src/test_runner/snap_test_runner.rs

```diff
@@ -17,15 +12,13 @@ pub struct SnapTestRunner<'a, I = ()> {
 }
 
 impl<'a> SnapTestRunner<'a> {
-    pub fn with_parser<PG, S>(name: &str, input: S) -> SnapTestRunner<'a, ()>
+    pub fn with_parser<S, PF>(name: &str, input: S, parser: PF) -> SnapTestRunner<'a, ()>
     where
-        PG: ParserGenerator,
         S: IntoSymbols<'a, Vec<Symbol<'a>>> + Clone + Into<&'a str>,
+        PF: for<'s> Fn(&'s [Symbol<'s>]) -> (String, &'s [Symbol<'s>]),
```

Comment:
This trait bound can be relaxed so that we accept `Fn`, `FnMut` and `FnOnce`: 

```suggestion
        PF: for<'s> FnOnce(&'s [Symbol<'s>]) -> (String, &'s [Symbol<'s>]),
```

See [docs on `FnOnce`](https://doc.rust-lang.org/std/ops/trait.FnOnce.html) for explanation.

Reply:
We do use the function in a loop at some point, so this is not possible.